python/processors/nft_orderbooks/parsers/souffle_parser.py

```diff
@@ -1,108 +1,280 @@
+    display_event_type = event_utils.get_event_type_short(event)
+    display_event_type = display_event_type.replace("<0x1::aptos_coin::AptosCoin>", "")
+
+    if display_event_type not in SOUFFLE_MARKETPLACE_EVENT_TYPES:
+        return []
+
+    data = json.loads(event.data)
+
+    # Collection, token, and creator parsing
+    token_data_id_struct = data.get("token_id", {}).get("token_data_id", {})
+    collection = token_data_id_struct.get("collection", None) or ""
+    token_name = token_data_id_struct.get("name", None) or ""
+    creator = token_data_id_struct.get("creator", None) or ""
+
+    token_data_id_type = TokenDataIdType(creator, collection, token_name)
+
+    # Price parsing
+    price = (
+        data.get("price")
+        or data.get("coin_per_token")
+        or data.get("coin_amount")
+        or None
+    )
+
+    # Token amount parsing
+    token_amount = int(data.get("token_amount", 1))
+
+    standard_event_type = standardize_marketplace_event_type(display_event_type)
+
+    # Buyer and seller parsing
+    buyer = data.get("buyer", None)
+    if standard_event_type == StandardMarketplaceEventType.LISTING_CANCEL:
+        seller = (
+            transaction_utils.get_sender(user_transaction) if user_transaction else None
         )
-        price = (float(price) / 10**8) if price != None else None
-
-        # Token amount parsing
-        token_amount = int(data.get("token_amount", 1))
-
-        standard_event_type = standardize_marketplace_event_type(display_event_type)
-
-        # Buyer and seller parsing
-        buyer = data.get("buyer", None)
-        if standard_event_type == StandardMarketplaceEventType.LISTING_CANCEL:
-            user_transaction = transaction_utils.get_user_transaction(transaction)
-            seller = (
-                transaction_utils.get_sender(user_transaction)
-                if user_transaction
-                else None
-            )
-        else:
-            seller = data.get("token_owner", None)
-
-        activity = NFTMarketplaceEvent(
-            transaction_version=event.transaction_version,
-            event_index=event.event_index,
-            event_type=display_event_type,
-            standard_event_type=standardize_marketplace_event_type(
-                display_event_type
-            ).value,
-            creator_address=standardize_address(creator),
+    else:
+        seller = data.get("token_owner", None)
+
+    activity = NFTMarketplaceEvent(
+        transaction_version=transaction_metadata.transaction_version,
+        event_index=event_index,
+        event_type=display_event_type,
+        standard_event_type=standardize_marketplace_event_type(
+            display_event_type
+        ).value,
+        creator_address=standardize_address(creator),
+        collection=token_data_id_type.get_collection_trunc(),
+        token_name=token_data_id_type.get_name_trunc(),
+        token_data_id=token_data_id_type.to_hash(),
```

Comment:
actually you might want to add '0x' in front of this to enable join w/ the rest of the data. To confirm check current_token_ownerships_v2

Reply:
The `to_hash` function calls `standardize_address` so the address will be `0x` prefixed. But good point, let me check that we're doing it correctly.

Reply:
Checked a random NFT listing - the token_data_id and seller address in the listing matches with a row in current_token_ownership_v2.

| current_token_ownerships_v2 | current_nft_listings | 
| --- | --- |
| ![Screenshot 2023-06-23 at 10 59 33 AM](https://github.com/aptos-labs/aptos-indexer-processors/assets/8248583/1bd206bc-f384-44f8-8b56-00fadb2b491b) | ![Screenshot 2023-06-23 at 10 59 51 AM](https://github.com/aptos-labs/aptos-indexer-processors/assets/8248583/2d13b0f5-66d0-4672-bf5a-9d98c716f93d) |